packages/vue/src/utils/add-linting.ts

```diff
@@ -0,0 +1,61 @@
+import { Tree } from 'nx/src/generators/tree';
+import { Linter, lintProjectGenerator } from '@nx/linter';
+import { joinPathFragments } from 'nx/src/utils/path';
+import { addDependenciesToPackageJson, runTasksInSerial } from '@nx/devkit';
+import { NormalizedSchema } from '../generators/library/schema';
+import { extraEslintDependencies } from './lint';
+import {
+  addExtendsToLintConfig,
+  isEslintConfigSupported,
+} from '@nx/linter/src/generators/utils/eslint-file';
+
+export async function addLinting(
+  host: Tree,
+  options: {
+    linter: Linter;
+    name: string;
+    projectRoot: string;
+    unitTestRunner?: 'jest' | 'vitest' | 'none';
+    setParserOptionsProject?: boolean;
+    skipPackageJson?: boolean;
+    rootProject?: boolean;
+  },
+  projectType: 'lib' | 'app'
+) {
+  if (options.linter === Linter.EsLint) {
```

Comment:
We probably want to call `lintProjectGenerator` here to setup and install any packages needed.